app/models/ext_management_system.rb

```diff
@@ -476,4 +481,15 @@ def stop_event_monitor_queue_on_credential_change
       self.stop_event_monitor_queue
     end
   end
+
+  def blacklisted_event_names
+    (
+      self.class.blacklisted_event_entries.pluck(:event_name) +
+      blacklisted_events.where(:enabled => true).pluck(:event_name)
+    ).uniq.sort
+  end
+
+  def self.blacklisted_event_entries
+    BlacklistedEvent.where(:enabled => true, :provider_model => name, :ems_id => nil)
```

Comment:
@lfu This method should be renamed `blackedlisted_events` and the `:enabled => true` should be left to the caller.  In this case that would be line 487 above.